src/cpp/desktop/DesktopMain.cpp

```diff
@@ -117,9 +117,7 @@ Error removeStaleOptionsLockfile()
 
 void initializeSharedSecret()
 {
-   sharedSecret = QString::number(rand())
-                  + QString::number(rand())
-                  + QString::number(rand());
+   sharedSecret = QString::fromStdString(core::system::generateUuid());
```

Comment:
On Windows, I was seeing `rand()` producing identical values in different machines when launched at the same time; this should be altogether safer.